### DISCUSSION TITLE
Unknown 140 events in Activity Files

### DISCUSSION THREAD

**Discussion by markbulk** [tcgoetz/GarminDB#106] [Show and tell]
I was able to get some useful information on the Garmin Developer forum for [the unknown 140 events](https://github.com/tcgoetz/GarminDB/wiki/Fit-Files).

See the [forum thread](https://forums.garmin.com/developer/fit-sdk/f/discussion/254469/list-of-undocumented-mesg_num).

I hope that this is useful.  Even more useful would be for Garmin to fully document their file format.

**Comment by tcgoetz** [comment]
Added some of the first beat message data: https://github.com/tcgoetz/Fit/commit/d3a93cd2478baa5ef82443e07273bdf617f375eb

**Comment by tcgoetz** [comment]
Added parsing for heart rate zones message: https://github.com/tcgoetz/Fit/commit/99dd293c48ef9691bece81a6e46937661a113098

**Comment by tcgoetz** [comment]
Thanks for finding the info for parsing these message types. Let me know if you find more.

I haven't added any of the newly parsed fields to the DB yet. Let me know if you have thoughts on how you will use it.

**Comment by hirael** [comment]
The above Garmin forum also said that "global message id 3, field 24, offset +1900 = year of birth"

**Reply by tcgoetz** [reply]
Added to develop branch.

**Comment by bombsimon** [comment]
Sorry for bringing this old discussion back to life but was reading the wiki and ended up on https://github.com/tcgoetz/GarminDB/wiki/Fit-Files.

@harryo has been doing an awesome work [summarizing undocumented types](https://medium.com/@harry_90407/beyond-the-fit-sdk-c9969430aeeb) and they can be found in [this](https://docs.google.com/spreadsheets/d/1x34eRAZ45nbi3U3GyANotgmoQfj0fR49wBxmL-oLogc/edit?gid=618517139#gid=618517139) Google Sheet.

Not sure if this is still relevant since 140 is in the wiki but apparently fixed already so if there are no more unknowns maybe the wiki page should be removed.

**Comment by tcgoetz** [comment]
I've started adding these addition data fields. Currently in the Fit module develop branch.
